pkg/azuredisk/controllerserver.go

```diff
@@ -1239,6 +1254,268 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	return azureutils.GetEntriesAndNextToken(req, snapshots)
 }
 
+func (d *Driver) CreateVolumeGroupSnapshot(ctx context.Context, req *csi.CreateVolumeGroupSnapshotRequest) (*csi.CreateVolumeGroupSnapshotResponse, error) {
+	sourceVolumeIDs := req.GetSourceVolumeIds()
+	if len(sourceVolumeIDs) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "CreateVolumeGroupSnapshot Source Volume IDs must be provided")
+	}
+	for _, sourceVolumeID := range sourceVolumeIDs {
+		if len(sourceVolumeID) == 0 {
+			return nil, status.Error(codes.InvalidArgument, "CreateVolumeGroupSnapshot Source Volume ID must be provided")
+		}
+	}
+	volumeGroupSnapshotName := req.Name
+	if len(volumeGroupSnapshotName) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume group snapshot name must be provided")
+	}
+
+	var customTags string
+	// set incremental snapshot as true by default
+	incremental := true
+	var resourceGroup, dataAccessAuthMode string
+	var err error
+	localCloud := d.cloud
+	tags := make(map[string]*string)
+
+	parameters := req.GetParameters()
+	for k, v := range parameters {
```

Comment:
`d.CreateSnapshot` also has all these parameter validation, what about make this function lightweight, make use of all the validation logic in `d.CreateSnapshot`

Reply:
done, just use these validation in `d.CreateSnapshot` is enough